src/mz-debug/src/kubectl_port_forwarder.rs

```diff
@@ -174,38 +180,81 @@ pub async fn find_environmentd_service(
 /// Returns Vec<(service_name, ports)> for cluster services
 pub async fn find_cluster_services(
     client: &Client,
-    k8s_namespaces: &Vec<String>,
+    k8s_namespace: &String,
+    mz_instance_name: &String,
 ) -> Result<Vec<ServiceInfo>> {
-    for namespace in k8s_namespaces {
-        let services: Api<Service> = Api::namespaced(client.clone(), namespace);
-        let services = services
-            .list(&ListParams::default())
-            .await
-            .with_context(|| format!("Failed to list services in namespace {}", namespace))?;
-        let cluster_services: Vec<ServiceInfo> = services
-            .iter()
-            .filter_map(|service| {
-                let name = service.metadata.name.clone()?;
-                let spec = service.spec.clone()?;
-                let selector = spec.selector?;
-                let ports = spec.ports?;
-
-                // Check if this is a cluster service
-                if selector.get("environmentd.materialize.cloud/namespace")? != "cluster" {
-                    return None;
-                }
+    let pods: Api<Pod> = Api::namespaced(client.clone(), k8s_namespace);
 
-                Some(ServiceInfo {
-                    service_name: name,
-                    service_ports: ports,
-                    namespace: namespace.clone(),
-                })
+    let pods_label_filter = format!(
+        "environmentd.materialize.cloud/namespace=cluster,materialize.cloud/organization-name={}",
+        mz_instance_name
+    );
+
+    let pods = pods
+        .list(&ListParams::default().labels(&pods_label_filter))
+        .await
+        .with_context(|| format!("Failed to list pods in namespace {}", k8s_namespace))?;
+
+    // Create a set of service IDs from pod labels
+    let service_ids_to_scrape: std::collections::BTreeSet<String> = pods
+        .iter()
+        .filter_map(|pod| {
+            pod.metadata
+                .labels
+                .as_ref()
+                .and_then(|labels| labels.get("environmentd.materialize.cloud/service-id"))
```

Comment:
I found a bug in our service labeling! We don't have any labels identifying which materialize instance a service should point to. This may cause services to go to the wrong materialize instance!

This also means you should probably change to getting the services with an owner reference pointing at environmentd.